iris-mpc-store/src/lib.rs

```diff
@@ -74,6 +76,43 @@ impl StoredIris {
     pub fn id(&self) -> i64 {
         self.id
     }
+
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, eyre::Error> {
+        let mut cursor = 0;
+
+        // Helper closure to extract a slice of a given size
+        let extract_slice =
+            |bytes: &[u8], cursor: &mut usize, size: usize| -> Result<Vec<u8>, eyre::Error> {
+                if *cursor + size > bytes.len() {
+                    return Err(eyre!("Exceeded total bytes while extracting slice",));
+                }
+                let slice = &bytes[*cursor..*cursor + size];
+                *cursor += size;
+                Ok(slice.to_vec())
+            };
+
+        // Parse `id` (i64)
+        let id_bytes = extract_slice(bytes, &mut cursor, 4)?;
+        let id = u32::from_be_bytes(
+            id_bytes
+                .try_into()
+                .map_err(|_| eyre!("Failed to convert id bytes to i64"))?,
+        ) as i64;
+
+        // parse codes and masks
+        let left_code = extract_slice(bytes, &mut cursor, 25_600)?;
+        let left_mask = extract_slice(bytes, &mut cursor, 12_800)?;
+        let right_code = extract_slice(bytes, &mut cursor, 25_600)?;
+        let right_mask = extract_slice(bytes, &mut cursor, 12_800)?;
```

Comment:
maybe use the `IRIS_CODE_LENGTH` and `MASK_CODE_LENGTH` here instead (multiplied by `mem::size_of<u16>()`)